cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/generator/GeneratorWrapperFactory.java

```diff
@@ -66,10 +67,49 @@ public static CobiGenWrapper createGenerator(ISelection selection, IProgressMoni
       throws GeneratorCreationException, GeneratorProjectNotExistentException, InvalidInputException {
 
     List<Object> extractedInputs = extractValidEclipseInputs(selection);
+    try {
+      if (extractedInputs.size() > 0) {
+        monitor.subTask("Initialize CobiGen instance");
+        CobiGen cobigen = initializeGenerator();
+
+        monitor.subTask("Reading inputs...");
+        monitor.worked(10);
+        Object firstElement = extractedInputs.get(0);
+
+        if (firstElement instanceof IJavaElement) {
+          LOG.info("Create new CobiGen instance for java inputs...");
+          return new JavaInputGeneratorWrapper(cobigen, ((IJavaElement) firstElement).getJavaProject().getProject(),
+              JavaInputConverter.convertInput(extractedInputs, cobigen), monitor);
+        } else if (firstElement instanceof IResource) {
+          LOG.info("Create new CobiGen instance for file inputs...");
+          return new FileInputGeneratorWrapper(cobigen, ((IResource) firstElement).getProject(),
+              FileInputConverter.convertInput(cobigen, extractedInputs), monitor);
+        }
+      }
+    } catch (GeneratorCreationException e) {
+      throw new DeprecatedMonolithicTemplatesException();
+    }
+    return null;
+  }
 
+  /**
+   * Creates a generator dependent on the input of the selection
+   *
+   * @param selection current {@link IStructuredSelection} treated as input for generation
+   * @param monitor tracking progress
+   * @return a specific {@link CobiGenWrapper} instance
+   * @throws GeneratorCreationException if any exception occurred during converting the inputs or creating the generator
+   * @throws GeneratorProjectNotExistentException if the generator configuration project does not exist
+   * @throws InvalidInputException if the selection includes non supported input types or is composed in a non supported
+   *         combination of inputs.
+   */
+  public static CobiGenWrapper createGeneratorWithMonolithicTemplates(ISelection selection, IProgressMonitor monitor)
+      throws GeneratorCreationException, GeneratorProjectNotExistentException, InvalidInputException {
+
+    List<Object> extractedInputs = extractValidEclipseInputs(selection);
     if (extractedInputs.size() > 0) {
       monitor.subTask("Initialize CobiGen instance");
-      CobiGen cobigen = initializeGenerator();
+      CobiGen cobigen = initializeGeneratorWithMonolithicTemplates();
```

Comment:
What about the previous method? Is there now a lot of redundant and dead code?

Reply:
The problem is that the Eclipse tests have to run with the monolithic configuration. I had to integrate the "postpone" button, such that the generation has to continue.  

Firstly the original method will be called then
the new copy of the original method has to call CobiGenFactory.create(....,true ) with the boolean allowing old configuration to continue working.